cmd/api/routers/routers.go

```diff
@@ -28,3 +28,18 @@ func (c *Controller) GetAllResources(context *gin.Context) {
 
 	context.JSON(http.StatusOK, resources)
 }
+
+func (c *Controller) GetNamespacedResources(context *gin.Context) {
+	group := context.Param("group")
+	version := context.Param("version")
+	namespace := context.Param("namespace")
+	// FIXME - Title is deprecated and this conversion must be done in a different way
+	resource := strings.Title(context.Param("resourceType")) //nolint:staticcheck
+	resources, err := c.Database.QueryNamespacedResources(context, resource[:len(resource)-1], group, version, namespace)
```

Comment:
Is `resource` a resource name? It looks like getting the resource name is going to be tricky for the `Database.QueryNamespacedResources` call. I suggest making a small function like `func GetResourceName(context *gin.Context)` to return the string that `QueryNamespacedResources` needs. That way you don't need to trim the name inside the `QueryNamespacedResources` call and you can create a unit test that verifies you are still getting the right string after you replace the deprecated `Title` code.

Reply:
Yes, this logic will be updated soon, but I extracted it to a different method.

I will probably add a request to the kubernetes cluster as a middleware function and set the singular resource name in the context, but I think that for now it makes more sense having it in a separated private function.

Already updated so I'll close the thread